src/Reflection/EntityFieldReflection.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace PHPStan\Reflection;
+
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\Type;
+
+/**
+ * Allows field access via magic methods
+ *
+ * See \Drupal\Core\Entity\ContentEntityBase::__get and ::__set.
+ */
+class EntityFieldReflection implements PropertyReflection
+{
+
+  /** @var ClassReflection */
+    private $declaringClass;
+
+  /** @var string */
+    private $propertyName;
+
+    public function __construct(ClassReflection $declaringClass, string $propertyName)
+    {
+        $this->declaringClass = $declaringClass;
+        $this->propertyName = $propertyName;
+    }
+
+    public function getType(): Type
+    {
+        if ($this->propertyName == 'original') {
+          // See Drupal\Core\Entity\EntityStorageBase::doPreSave
+            return new ObjectType('Drupal\Core\Field\FieldItemListInterface');
```

Comment:
Isn't `original` a copy of the entity, so it should be an entity interface?

```
    // Load the original entity, if any.
    if ($id_exists && !isset($entity->original)) {
      $entity->original = $this->loadUnchanged($id);
    }
```

```
  /**
   * Loads an unchanged entity from the database.
   *
   * @param mixed $id
   *   The ID of the entity to load.
   *
   * @return \Drupal\Core\Entity\EntityInterface|null
   *   The unchanged entity, or NULL if the entity cannot be loaded.
   *
   * @todo Remove this method once we have a reliable way to retrieve the
   *   unchanged entity from the entity object.
   */
  public function loadUnchanged($id);
```

This should be `\Drupal\Core\Entity\EntityInterface`

Reply:
seems like a typo, because it returns FieldItemListInterface outside the if anyhow